src/libs/OptionsListUtils.js

```diff
@@ -396,6 +397,12 @@ function getLastMessageTextForReport(report) {
         lastMessageTextFromReport = ReportUtils.getReportPreviewMessage(iouReport, lastReportAction);
     } else if (ReportActionUtils.isModifiedExpenseAction(lastReportAction)) {
         lastMessageTextFromReport = ReportUtils.getModifiedExpenseMessage(lastReportAction);
+    } else if (
+        lastActionName === CONST.REPORT.ACTIONS.TYPE.TASKCOMPLETED ||
+        lastActionName === CONST.REPORT.ACTIONS.TYPE.TASKREOPENED ||
+        lastActionName === CONST.REPORT.ACTIONS.TYPE.TASKCANCELLED
```

Comment:
@rushatgabhane 

On referencing the code [here](https://github.com/Expensify/App/blob/d61f55ac16b797f831e4cb1b9669ba326ac7b8f9/src/libs/actions/Task.js#L924C1-L945C2), it does not look like `TASKEDITED` is an action that has a message associated with it. Also, `TASKEDITED` is not a visible action like the other task actions as it can be seen [here](https://github.com/Expensify/App/blob/d61f55ac16b797f831e4cb1b9669ba326ac7b8f9/src/pages/home/report/ReportActionItem.js#L316C1-L327C10)

It seems that `TASKEDITED` is not similar to other task actions and maybe by design. However, we have added support for `TASKCANCELLED` and `TASKREOPENED` along with `TASKCOMPLETED` since they all behave similarly. 

So, I think we should not consider `TASKEDITED` here as it looks outside of the scope in our current context. What do you think?

Reply:
@rushatgabhane 
I just checked the contents of the action message (ref below). 
There are two array elements in `message` for `TASKEDITED` unlike other task actions.
We also need to know how to construct the message here. And, it is not similar to other actions which has a single clear message. Let me know how we can proceed here.
![Screenshot 2023-09-27 at 1 09 18 PM](https://github.com/Expensify/App/assets/3004659/e72a00fd-695b-43d2-9258-cfa1c8cdaf55)

Reply:
@rushatgabhane 

Oh! How did I miss this?
Actually, `shouldReportActionBeVisible` is skipping `TASKEDITED` [here](https://github.com/Expensify/App/blob/41ddd7428340fd8d347e8160980809e76a93899b/src/libs/ReportActionsUtils.js#L345C1-L347C6)
Just found this out on investigating further after adding `TASKEDITED` along with other conditions.
This definitely doesn’t look like we should be handling here.

Reply:
> The existing code should work, right? `message[0].text`
> 
> I confirmed it locally yesterday

The existing code is supposed to return only 'You'. But, that too, only if `shouldReportActionBeVisible` returns true which is not the case here.

Reply:
> I confirmed it locally yesterday

@rushatgabhane 

Can you please share the details of your tests? 
We have an obvious disconnect here. So, just want to check if I am missing something very simple.

Reply:
@rushatgabhane 

Ah! Now, I get it.
Please have a look at the attached test video.
As seen in video, the BE doesn't update the text `message[0].text` with edited task name. 
Looks like a bug in BE.
Does this make sense?

https://github.com/Expensify/App/assets/3004659/e6201fb3-302b-488b-8db9-23dea44983b4